cropper/src/main/java/com/canhub/cropper/CropImage.java

```diff
@@ -210,7 +210,13 @@ public static Intent getPickImageChooserIntent(
             allIntents.addAll(getCameraIntents(context, packageManager));
         }
 
-        allIntents.add(getGalleryIntent(Intent.ACTION_GET_CONTENT, includeDocuments));
+        List<Intent> galleryIntents =
+                getGalleryIntents(packageManager, Intent.ACTION_GET_CONTENT, includeDocuments);
+        if (galleryIntents.size() == 0) {
+            // if no intents found for get-content try pick intent action (Huawei P9).
+            galleryIntents = getGalleryIntents(packageManager, Intent.ACTION_PICK, includeDocuments);
```

Comment:
What is the Huawei P9 issue? could you pointed me to some information or show the test you did?

Reply:
Honestly, I have no idea about that. I have never used a huawei device. Just the code from the legacy lib.